lib/membrane_opus/parser.ex

```diff
@@ -219,4 +216,21 @@ defmodule Membrane.Opus.Parser do
     present_frames = frame_lengths |> Enum.count(fn length -> length > 0 end)
     present_frames * frame_duration
   end
+
+  defp validate_pts_integrity(true = _check_pts_integrity?, packets, input_pts) do
+    cond do
+      length(packets) >= 2 and Enum.at(packets, 1).pts > input_pts ->
+        Membrane.Logger.warning("PTS values are overlapping")
+
+      length(packets) >= 2 and Enum.at(packets, 1).pts < input_pts ->
+        Membrane.Logger.warning("PTS values are not continous")
+
+      true ->
+        :ok
+    end
+  end
```

Comment:
```suggestion
  defp validate_pts_integrity(packets, input_pts) do
    cond do
      length(packets) < 2 or Enum.at(packets, 1).pts == input_pts -> 
        :ok
        
      Enum.at(packets, 1).pts > input_pts ->
        Membrane.Logger.warning("PTS values are overlapping")
        :ok

      Enum.at(packets, 1).pts < input_pts ->
        Membrane.Logger.warning("PTS values are not continous")
        :ok
    end
  end
```

1) I would avoid calling `length(packets)` two times, because this function has linear complexity
2) Move it to `Util` module, to stop reapeating yourself and delete this function from `Encoder`
3) IMO it would be prettier to remove flag from function args and wrap function call into an `if` statement, eg. `if check_pts_integrity?, do: Util.validate..`